drivers/periph_common/eeprom.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2018 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers
+ * @{
+ *
+ * @file
+ * @brief       Common eeprom functions implementation
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <string.h>
+#include "cpu.h"
+#include "assert.h"
+
+/* guard this file, must be done before including periph/eeprom.h */
+#if defined(EEPROM_SIZE)
+
+#include "periph/eeprom.h"
+
+void eeprom_read(uint32_t pos, uint8_t *data, size_t len)
```

Comment:
following my comment above having a proper return value, this could be something like:

```
if ((pos < EEPROM_SIZE) && (len < EEPROM_SIZE)) {
    int rc;
    for (rc = 0; (rc < len) && (pos + rc < EEPROM_SIZE); ++rc) {
        data[rc] = eeprom_read_byte(pos + rc);
    } 
    return rc;
}
return (-1);
```

applies for write function below too

Reply:
Not the most readable. :smirk: 

Maybe just copy "len" and return it?